.github/workflows/ci.yml

```diff
@@ -0,0 +1,65 @@
+# Run basic tests for this app on the latest aiidalab-docker image.
+
+name: continuous-integration
+
+on:
+  [push]
+
+jobs:
+
+  build-docker-image:
+
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+
+    outputs:
+      image: ${{ steps.tag.outputs.image }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: build image
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          tag_with_ref: true
+          tag_with_sha: true
+          cache_froms: "aiidalab/aiidalab-docker-stack:latest,aiidalab/aiidalab-docker-stack:stable,aiidateam/aiida-core:latest"
+
+      - name: set output image
+        id: tag
+        run: |
+          IMAGE_REF=`echo ${{ github.sha }} | sed 's/^\(.......\).*/\1/'`
+          echo "::set-output name=image::${{ github.repository }}:sha-${IMAGE_REF}"
+
+  test-apps:
+
+    runs-on: ubuntu-latest
+    needs: build-docker-image
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        app: [
+            # aiidalab/aiidalab-home,  -- failing
+            aiidalab/aiidalab-hello-world,
+            aiidalab/aiidalab-widgets-base,
+            aiidalab/aiidalab-qe,
+            # aiidalab/aiidalab-optimade, -- failing
+          ]
+        browser: [ chrome, firefox ]
+      fail-fast: false
+
+    steps:
+
+      - name: Checkout app
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ matrix.app }}
+
+      - name: Test app
+        uses: aiidalab/aiidalab-test-app-action@v2
```

Comment:
would you mind adding one line of comment here on what this actually tests?

Reply:
I've generally overhauled the documentation of the workflow definition.